src/Core/Config/PreloadConfigAdapter.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Friendica\Core\Config;
+
+use dba;
+use Exception;
+use Friendica\BaseObject;
+
+require_once 'include/dba.php';
+
+/**
+ * Preload ConfigAdapter
+ *
+ * Minimize the number of database queries to retrieve configuration values at the cost of memory.
+ *
+ * @author Hypolite Petovan <mrpetovan@gmail.com>
+ */
+class PreloadConfigAdapter extends BaseObject implements IConfigAdapter
+{
+	private $config_loaded = false;
+
+	public function __construct()
+	{
+		$this->load();
+	}
+
+	public function load($family = 'config')
+	{
+		if ($this->config_loaded) {
+			return;
+		}
+
+		$a = self::getApp();
+
+		$configs = dba::select('config', ['cat', 'v', 'k']);
+		while ($config = dba::fetch($configs)) {
+			$cat   = $config['cat'];
+			$k     = $config['k'];
+			$value = (preg_match("|^a:[0-9]+:{.*}$|s", $config['v']) ? unserialize($config['v']) : $config['v']);
+
+			if ($cat === 'config') {
+				$a->config[$k] = $value;
+			} else {
+				$a->config[$cat][$k] = $value;
+			}
+		}
+		dba::close($configs);
+
+		$this->config_loaded = true;
+	}
+
+	public function get($cat, $k, $default_value = null, $refresh = false)
+	{

```

Comment:
```$refresh = true``` has to be coded as well. There are places in the code that depend on always fetching the latest config value.

For example "src/Util/ExAuth.php" is a long running process - it is started with an ejabberd and then runs for several hours/days/weeks.

Same goes with "scripts/worker.php". It has to check whether the system is in maintenance mode. Then it has to stop immediately. (BTW: There is currently a bug in the worker)

Reply:
I see, thanks for your vigilance.

Reply:
Done.